tests/datamodules/test_experience_sources.py

```diff
@@ -32,123 +39,285 @@ def test_iterator(self):
             break
 
 
-class TestExperienceSource(TestCase):
+class TestBaseExperienceSource(TestCase):
 
     def setUp(self) -> None:
         self.net = Mock()
         self.agent = DummyAgent(net=self.net)
         self.env = gym.make("CartPole-v0")
         self.device = torch.device('cpu')
-        self.source = ExperienceSource(self.env, self.agent)
-
-    def test_step(self):
-        exp, reward, done = self.source.step(self.device)
-        self.assertEqual(len(exp), 5)
+        self.source = DummyExperienceSource(self.env, self.agent)
 
-    def test_episode(self):
-        total_reward = self.source.run_episode(self.device)
-        self.assertIsInstance(total_reward, float)
+    def test_dummy_base_class(self):
+        """Tests that base class is initialized correctly"""
+        self.assertTrue(isinstance(self.source.env, gym.Env))
+        self.assertTrue(isinstance(self.source.agent, Agent))
+        out = next(iter(self.source))
+        self.assertTrue(torch.all(out.eq(torch.ones(3))))
 
 
-class TestNStepExperienceSource(TestCase):
+class TestExperienceSource(TestCase):
 
     def setUp(self) -> None:
         self.net = Mock()
         self.agent = DummyAgent(net=self.net)
-        self.env = gym.make("CartPole-v0")
-        self.n_step = 2
-        self.source = NStepExperienceSource(self.env, self.agent, n_steps=self.n_step)
+        self.env = [gym.make("CartPole-v0") for _ in range(2)]
+        self.device = torch.device('cpu')
+        self.source = ExperienceSource(self.env, self.agent)
+
+    def test_init(self):
+        """Test that experience source is setup correctly"""
+        self.assertEqual(self.source.n_steps, 1)
+        self.assertIsInstance(self.source.pool, list)
+
+        self.assertEqual(len(self.source.states), len(self.source.pool))
+        self.assertEqual(len(self.source.histories), len(self.source.pool))
+        self.assertEqual(len(self.source.cur_rewards), len(self.source.pool))
+        self.assertEqual(len(self.source.cur_steps), len(self.source.pool))
+
+    def test_init_single_env(self):
+        """Test that if a single env is passed that it is wrapped in a list"""
+        single_env = Mock()
```

Comment:
you already have it as `self.net`

Reply:
im not sure what you mean

Reply:
that in the test init you already have `self.net = Mock()` so here you can use `self.net` instead of `single_env`